Update chapter list

adamwoodnz opened the pull request:
Closes #265, #310, #313

Updates styling of chapter list:
- Font weights, colors, list item styles

Note: Does not include #306 

Note also that the root handbook page is always shown in the current prod chapter list, so 'Best Practices' being shown on the WPCS page in the screenshots below is consistent with the other handbooks.

### Screenshots

#### Coding standards
![localhost_8888_coding-standards_wordpress-coding-standards_(Desktop)](https://github.com/WordPress/wporg-developer/assets/1017872/bcc9ba02-9830-4be8-bb68-7724767c2852)

#### Block Editor

![localhost_8888_block-editor_reference-guides_theme-json-reference_theme-json-living_(Desktop)](https://github.com/WordPress/wporg-developer/assets/1017872/fa98cd0a-79df-4d1a-b507-58fcc5166e8f)

#### Advanced admin

![localhost_8888_advanced-administration_security_backup_(Desktop)](https://github.com/WordPress/wporg-developer/assets/1017872/78d67509-91b8-4d4d-a5fc-837be3bb5370)

adamwoodnz committed:
Restyle chapter list

adamwoodnz committed:
Revert updating chevron image

jasmussen commented:
Nice, this is a big win. I think it can in principle land as is. 

That said, per a recent revisit to this section, some opportunities for simplification were surfaced. In the case of the chapters list, it was mainly some small things:

![Screenshot 2023-10-30 at 10 21 05](https://github.com/WordPress/wporg-developer/assets/1204802/6fbcd4a5-fdf9-49f0-9d24-99999a7df90e)

* Smaller chevron (chevron-small) from [the WordPress Design Library](https://www.figma.com/file/e4tLacmlPuZV47l7901FEs/WordPress-Design-Library?type=design&node-id=8%3A13&mode=design&t=d3BLmBFQbGJsdwzC-1). 
* Chevron pointing right means collapsed, pointing down means open
* 3x3 dot instead of 4x4 dot
* Small margin tweaks.

You can dive into the [Figma here](https://www.figma.com/file/2WxlJFzMJvqPfZL1EkAOVp/Developer-Resources?type=design&node-id=3735%3A18466&mode=design&t=aNq0YJwNFQZwW2YV-1).

---

I noted some larger changes as well, which I'll show here for reference. I wouldn't call any of them urgent or blocking, but they simplify the overall appearance a bit, and could be applied across any section that uses chapters or tables of content, including Documentation.

![Expanded, collapsed, scrolled](https://github.com/WordPress/wporg-developer/assets/1204802/3261253b-3bc4-467e-ace7-43e71d548f8d)

![Responsive](https://github.com/WordPress/wporg-developer/assets/1204802/8c4a533f-a9a0-4696-b4ed-8f09bf8b549d)

Notably the removal of dots and the background from the TOC. Both scroll with you down the page. "Back to top" can appear as you scroll a bit. Most of those changes are in the "enhancements" category and should not block anything, or even need be part of a V1. I can create issues separately if need be, but wanted your take on it first.

---

Also a think for the future, and IMO also in the nice-to-have category is the expand/collapse behavior of the left sidebar. Here's the SVG for the new icon for that, which has also been added to the WordPress Design Library:

```
<svg width="24" height="24" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M18 17.5v-11h-1.5v11H18Z" fill="#1E1E1E"/><path d="m6 12 4.25-4.5M6 12h9m-9 0 4.25 4.5" stroke="#1E1E1E" stroke-width="1.5"/></svg>
```

adamwoodnz commented:
Thanks for the feedback @jasmussen !

> In the case of the chapters list, it was mainly some small things

👍 I'll make these changes in this PR I think

> I can create issues separately if need be

Yes please, a new issue in [wporg-mu-plugins](https://github.com/WordPress/wporg-mu-plugins) for the ToC changes. I might do the background and bullet changes quickly, then follow up with the rest.

> Also a think for the future, and IMO also in the nice-to-have category is the expand/collapse behavior of the left sidebar

👍 I've got a separate issue for this #306 

I'll also tighten up the column spacing, but that doesn't need an issue.

adamwoodnz committed:
Adjust responsive layout of handbooks

- Reduce width of chapter list and ToC
- Retain column layout for chapter list and content on tablet

adamwoodnz committed:
Update chapter list style to match new design

Smaller icons and tighter spacing

adamwoodnz commented:
Chapter list and responsive layout updated:

### Screenshots

#### Monitor

![localhost_8888_advanced-administration_server_web-server_httpd_(Monitor) (1)](https://github.com/WordPress/wporg-developer/assets/1017872/100b2a84-7540-4688-a475-c0733549df02)

| Desktop | Tablet | Mobile |
|-|-|-|
| ![localhost_8888_advanced-administration_server_web-server_httpd_(Desktop) (1)](https://github.com/WordPress/wporg-developer/assets/1017872/26962302-0e2e-4376-9ea7-75cd5beece2a) | ![localhost_8888_advanced-administration_server_web-server_httpd_(iPad)](https://github.com/WordPress/wporg-developer/assets/1017872/13ae78a1-2b73-4be6-bd4a-3b943d90d5df) | ![localhost_8888_advanced-administration_server_web-server_httpd_(Samsung Galaxy S20 Ultra) (1)](https://github.com/WordPress/wporg-developer/assets/1017872/72883867-6c16-4877-90a6-40a28b86146b) |

adamwoodnz committed:
Fix overflow of article content on smaller screens

adamwoodnz committed:
Update left padding to match new design
